mundane-bytes/src/test/scala/com/ambiata/mundane/bytes/BufferSpec.scala

```diff
@@ -0,0 +1,47 @@
+package com.ambiata.mundane.bytes
+
+import com.ambiata.disorder._
+import org.specs2.{ScalaCheck, Specification}
+
+class BufferSpec extends Specification with ScalaCheck { def is = s2"""
+
+  $wrapArray
+  $wrapArrayAndEmpty
+  $allocateNoResize
+  $allocateResize
+  $allocateHasCapacity
+  $allocateCopy
+"""
+
+  // TODO Add offset change and then allocate!
```

Comment:
I think if you add a way to construct a "populated" array, a bunch of other props will fall out (i.e. big array with some offset and length, a slice operation that slices out a precise length array == original subset, or similar).